bench/utils.py

```diff
@@ -1053,14 +1053,18 @@ def generate_command_cache(bench_path='.'):
 		os.remove(bench_cache_file)
 
 	try:
-		output = get_cmd_output("{0} -m frappe.utils.bench_helper get-frappe-commands".format(python), cwd=sites_path)
+		command = "{0} -m frappe.utils.bench_helper get-frappe-commands".format(python)
+		logger.debug('generate_command_cache(\'%s\') executing: %s', bench_path, command)
+		output = get_cmd_output(command, cwd=sites_path)
 		with open(bench_cache_file, 'w') as f:
 			json.dump(eval(output), f)
 		return json.loads(output)
 
 	except subprocess.CalledProcessError as e:
+		logger.error('generate_command_cache(\'%s\') failed executing: %s', bench_path, command, exc_info=e)
 		if hasattr(e, "stderr"):
 			print(e.stderr.decode('utf-8'))
+		raise e
```

Comment:
What is the reason for raising the error?

Reply:
generate_command_cache() is used by cli's get_frappe_commands(). Previous behavior ignores errors during call of /home/frappe/frappe-bench/env/bin/python -m frappe.utils.bench_helper get-frappe-commands, which breaks following code and results in confusing error messages.

By not throwing exception (previous behavior), execution will continue but will fail anyway later, with even more confusing/unpredictable error message that is hard to diagnose.

Reply:
Hey @ceefour, we want failures under `generate_command_cache` to be silent and non-blocking. 

So, I believe we shouldn't raise here, rather fix `get_frappe_commands` in `cli.py` to make sure it always returns a list, empty or filled. In that case, the screenshot with the error you shared earlier wouldn't arise.

Sorry for the delay in response.